examples/lightning/async_checkpoint_writing.py

```diff
@@ -0,0 +1,42 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+from lightning import Trainer
+from lightning.pytorch.callbacks import ModelCheckpoint
+from lightning.pytorch.demos import WikiText2, LightningTransformer
+from lightning.pytorch.plugins import AsyncCheckpointIO
+from torch.utils.data import DataLoader
+
+from s3torchconnector.lightning import S3LightningCheckpoint
+
+
+def main(region: str, checkpoint_path: str):
+    dataset = WikiText2()
+    dataloader = DataLoader(dataset, num_workers=3)
+
+    model = LightningTransformer(vocab_size=dataset.vocab_size)
+    s3_lightning_checkpoint = S3LightningCheckpoint(region)
+    async_checkpoint = AsyncCheckpointIO(s3_lightning_checkpoint)
+
+    checkpoint_callback = ModelCheckpoint(
```

Comment:
Can we perhaps add some explanatory comments such as:
- checkpoint path will look like ... 
- we expect N checkpoints to be saved ... etc.

Same for the other examples.